sdk/servicebus/Azure.Messaging.ServiceBus/tests/Sender/ServiceBusMessageBatchTests.cs

```diff
@@ -104,6 +104,73 @@ public void DisposeIsDelegatedToTheTransportClient()
             Assert.That(mockBatch.DisposeInvoked, Is.True);
         }
 
+        /// <summary>
+        ///   Verifies property accessors for the <see cref="AmqpMessageBatch.TryAdd" />
+        ///   method.
+        /// </summary>
+        ///
+        [Test]
+        public void TryAddRespectsTheBatchLock()
+        {
+            var mockBatch = new MockTransportBatch();
+            var batch = new ServiceBusMessageBatch(mockBatch);
+            var message = new ServiceBusMessage(new byte[] { 0x21 });
+
+            Assert.That(batch.TryAdd(new ServiceBusMessage(new byte[] { 0x21 })), Is.True, "The event should have been accepted before locking.");
+
+            batch.Lock();
+            Assert.That(() => batch.TryAdd(new ServiceBusMessage(Array.Empty<byte>())), Throws.InstanceOf<InvalidOperationException>(), "The batch should not accept events when locked.");
+
+            batch.Unlock();
+            Assert.That(batch.TryAdd(new ServiceBusMessage(Array.Empty<byte>())), Is.True, "The event should have been accepted after unlocking.");
+        }
+
+        /// <summary>
+        ///   Verifies property accessors for the <see cref="AmqpMessageBatch.Clear" />
+        ///   method.
+        /// </summary>
+        ///
+        [Test]
+        public void ClearRespectsTheBatchLock()
+        {
+            var mockBatch = new MockTransportBatch();
+            var batch = new ServiceBusMessageBatch(mockBatch);
+            var eventData = new ServiceBusMessage(new byte[] { 0x21 });
+
+            Assert.That(batch.TryAdd(new ServiceBusMessage(new byte[] { 0x21 })), Is.True, "The event should have been accepted before locking.");
+
+            batch.Lock();
+            Assert.That(() => batch.Clear(), Throws.InstanceOf<InvalidOperationException>(), "The batch should not accept events when locked.");
+
+            batch.Unlock();
+            batch.Clear();
+
+            Assert.That(mockBatch.ClearInvoked, Is.True, "The batch should have been cleared after unlocking.");
+        }
+
+        /// <summary>
+        ///   Verifies property accessors for the <see cref="AmqpMessageBatch.Clear" />
+        ///   method.
+        /// </summary>
+        ///
+        [Test]
+        public void DisposeRespectsTheBatchLock()
+        {
+            var mockBatch = new MockTransportBatch();
+            var batch = new ServiceBusMessageBatch(mockBatch);
+            var eventData = new ServiceBusMessage(new byte[] { 0x21 });
```

Comment:
We are not using `eventData` in this method. Can we remove this?

Reply:
Yup, good call.